### PR TITLE
[FIX] various deep links issues

### LINKED ISSUE #3525: Open deep link duplicate 5 or 9 tab
**Describe the bug**
im open deep link but metamask duplicate 5 tab or 9 tab, should be open 1 tab only

**Screenshots**
Screenshot_2021-12-27-17-52-01-145_io metamask

Screenshot_2021-12-27-17-52-48-315_io metamask

**To Reproduce**
1. Generate deep link from [](https://metamask.github.io/metamask-deeplinks/)
2. Open deep link

**Smartphone (please complete the following information):**
 - Device: Android / IOS
 - OS: Android 11 / IOS 15
 - App Version: 3.8.0

### PR DESCRIPTION
**Pull request by sethkfman**
**Description**

There was a bug that allowed the creation of multiple Branch subscribers which would trigger multiple WalletConnect & Deep Link sessions. 

This also fixes various deeplinks related issues (also from WalletConnect).

Bumps react-native-branch to latest version (5.2.1)

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

**Issue**

Resolves #3525, #3534, #3646, #3395

### CODE REVIEW

**Review by gantunesr — APPROVED** [review]
LGTM

**Comment by Fatxx** [comment]
LGTM

**Review by gantunesr — APPROVED** [review]
Alt Text

**Review by atifbuttgss — APPROVED** [review]

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -51,10 +51,14 @@
 				<action android:name="android.intent.action.VIEW" />
 				<category android:name="android.intent.category.DEFAULT" />
 				<category android:name="android.intent.category.BROWSABLE" />
-				<data android:scheme="https" android:host="metamask.app.link" />
-				<data android:scheme="https" android:host="metamask-alternate.app.link" />
-				<data android:scheme="https" android:host="metamask.test-app.link" />
-				<data android:scheme="https" android:host="metamask-alternate.test-app.link" />
+				<data android:scheme="https" />
```
Comment (rickycodes) [inline android/app/src/main/AndroidManifest.xml:54]: this looks like a mistake? you're missing `android:host` for these. if it's intentional you probably only need the one tag?

File: app/components/Nav/App/index.js [hunk 2]
```diff
@@ -35,6 +35,8 @@ import { checkedAuth } from '../../../actions/user';
 import { setCurrentRoute } from '../../../actions/navigation';
 import { findRouteNameFromNavigatorState } from '../../../util/general';
 
+const isIos = Platform.OS === 'ios';
```
Comment (rickycodes) [inline app/components/Nav/App/index.js:38]: I'm seeing this a fair bit, there's actually a utility for this already in `util/device`: https://github.com/MetaMask/metamask-mobile/blob/develop/app/util/device/index.js#L15

**Review by rickycodes — CHANGES_REQUESTED** [review]
a few small suggestions

File: app/components/Nav/App/index.js (hunk 2 above)
Comment (andreahaku) [reply app/components/Nav/App/index.js:38]: awesome! thanks. Will use it.

**Review by rickycodes — APPROVED** [review]

**Pull request merged**